### PR TITLE
chore: optimize Tailwind CSS classes

### PR DESCRIPTION
**Pull request by wa0x6e**
### Summary

Fix and cleanup css, in anticipation of adding eslint rules for scss.

Those are just regular and simple fixes, without any controversial rules

- Remove invalid `button-outline` classes, ported from snapshot v1
- Remove invalid `text-color`
- Use shorthand classes when possible (`w-[] z-[]` to `size-[]`)
- Use tailwindcss own size class instead of arbitrary px values (`s-[4px]` to `s-1`)
- Arbitrary classes should not start by `-`, use negative value instead (`-mb-[5px]` will become `mb-[-5px]`) 
- Replace invalid class by their nearest valid equivalent (`flex-grow` -> `grow`, `z-5` -> `z-10`, etc ...)
- Remove conflicting class (`flex` and `box` simultaneous usage)

Todo in a later pr will be more heavy changes, like css classes ordering

### CODE REVIEW

**Review by Sekhmet — CHANGES_REQUESTED** [review]
nACK

Let's discuss each change first and submit smaller PRs for different changes.

As of now I think those points can have been submitted already:
- Remove conflicting class (flex and box simultaneous usage)
- Use shorthand classes when possible (w-[] z-[] to size-[])
- Replace invalid text-color class by text-skin-text
- Remove invalid button-outline classes, ported from snapshot v1
- Arbitrary classes should not start by -, use negative value instead (-mb-[5px] will become mb-[-5px])

Ideally those should be separated into each own PR so we can easily review it and test it doesn't break anything.

The rest should be discussed first.

File: apps/ui/src/App.vue [hunk 1]
```diff
@@ -78,11 +78,7 @@ watch(isSwiping, () => {
 </script>
 
 <template>
-  <div
-    ref="el"
-    class="min-h-screen"
-    :class="{ 'overflow-clip': scrollDisabled }"
-  >
+  <div ref="el" class="min-h-screen" :class="{ 'text-clip': scrollDisabled }">
```
[the remark this answers is not in the document]
Comment (wa0x6e) [reply apps/ui/src/App.vue:81]: This class has been renamed in tailwindcss 3 => https://github.com/snapshot-labs/snapshot-hub/blob/8e7e0a57689482ef52868f4fb875f3718ec183d0/src/helpers/keycard.ts#L11

[the remark this answers is not in the document]
Comment (Sekhmet) [reply apps/ui/src/App.vue:81]: Looks like you posted wrong link, but those two properties are different:
https://tailwindcss.com/docs/overflow

`overflow-clip`: -> `overflow: clip`.
`text-clip`: -> `text-overflow: clip`.

[the remark this answers is not in the document]
Comment (wa0x6e) [reply apps/ui/src/App.vue:81]: This is the correct link https://tailwindcss.com/docs/upgrade-guide#overflow-clip-ellipsis

[the remark this answers is not in the document]
Comment (wa0x6e) [reply apps/ui/src/App.vue:81]: But I agree with you, seems like the old one still is used. will revert this one

File: apps/ui/src/components/App/Sidebar.vue [hunk 2]
```diff
@@ -7,13 +7,13 @@ const followedSpacesStore = useFollowedSpacesStore();
 
 <template>
   <div
-    class="w-[72px] flex flex-col border-r fixed left-0 top-0 bottom-0 text-center h-screen"
+    class="w-[72px] flex flex-col border-r fixed left-0 inset-y-0 text-center h-screen"
   >
     <router-link :to="{ name: 'landing' }" class="h-[72px] block">
-      <IH-stop class="inline-block my-4 w-[32px] h-[32px] text-skin-link" />
+      <IH-stop class="inline-block my-4 size-5 text-skin-link" />
```
[the remark this answers is not in the document]
Comment (wa0x6e) [reply apps/ui/src/components/App/Sidebar.vue]: Will revert to keep everything aside from spacing to user arbitrary px values

**Comment by wa0x6e** [comment]
> nACK
> 
> Let's discuss each change first and submit smaller PRs for different changes.
> 
> As of now I think those points can have been submitted already:
> 
> * Remove conflicting class (flex and box simultaneous usage)
> * Use shorthand classes when possible (w-[] z-[] to size-[])
> * Replace invalid text-color class by text-skin-text
> * Remove invalid button-outline classes, ported from snapshot v1
> * Arbitrary classes should not start by -, use negative value instead (-mb-[5px] will become mb-[-5px])
> 
> Ideally those should be separated into each own PR so we can easily review it and test it doesn't break anything.
> 
> The rest should be discussed first.

Updated to revert all changes requiring any discussions.

This PR was meant to be a no-brainer changes, and should include only obvious fixes which does not require any testing

**Comment by Sekhmet** [comment]
For `text-color` -> `text-skin-text` changes - as `text-color` wasn't doing anything before shouldn't we just remove this property entirely? Now we either add property that doesn't change anything or we change colour making it break old behaviour.
In this case it looks like we just add property that doesn't actually do anything.

**Comment by wa0x6e** [comment]
> For `text-color` -> `text-skin-text` changes - as `text-color` wasn't doing anything before shouldn't we just remove this property entirely? Now we either add property that doesn't change anything or we change colour making it break old behaviour. In this case it looks like we just add property that doesn't actually do anything.

removed

File: apps/ui/src/components/ProposalResults.vue [hunk 3]
```diff
@@ -196,7 +196,7 @@ const otherResultsSummary = computed(() => {
       class="h-full flex items-center"
     >
       <div
-        class="rounded-full h-1.5 overflow-hidden"
+        class="rounded-full h-[4px].5 overflow-hidden"
```
Comment (Sekhmet) [inline apps/ui/src/components/ProposalResults.vue:199]: Is this intended change? Looks broken.

File: apps/ui/src/style.scss [hunk 4]
```diff
@@ -304,7 +304,7 @@ input:placeholder-shown {
   .s-label:has(+ textarea),
   .s-base:has(textarea:only-child) {
      &::before {
-      @apply block bg-gradient-to-b from-skin-border to-transparent h-2 w-full absolute top-[20px] left-0;
+      @apply block bg-gradient-to-b from-skin-border to-transparent h-[8px] w-full absolute top-3.5 left-0;
```
Comment (Sekhmet) [inline apps/ui/src/style.scss:307]: Not sure why `h-2` was changed to `h-[8px]`.

Comment (wa0x6e) [reply apps/ui/src/style.scss:307]: For consistency, everything that is not spacing related (`(m|p|space|gap)-[0-9]`) is using arbitrary pixels value.

we're already using `h-[32px]` for avatar, etc ...

Or should we keep using both h-2 and h-[8px] ?

Comment (Sekhmet) [reply apps/ui/src/style.scss:307]: I see, let's keep it like this then.

**Review by Sekhmet — APPROVED** [review]
tACK

**Comment by Sekhmet** [comment]
@wa0x6e merge conflict needs to be resolved.

**Pull request merged**